gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Api.java

```diff
@@ -0,0 +1,583 @@
+   * @param logName Required. The resource name of the log to delete.  Example:
+   * `"projects/my-project/logs/syslog"`.
+   */
+  public void deleteLog(String logName) {
+    DeleteLogRequest request =
+        DeleteLogRequest.newBuilder()
+        .setLogName(logName)
+        .build();
+
+    deleteLog(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Deletes a log and all its log entries.
+   * The log will reappear if it receives new entries.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param request The request object containing all of the parameters for the API call.
+   */
+  public void deleteLog(DeleteLogRequest request) {
+    deleteLogCallable().call(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Deletes a log and all its log entries.
+   * The log will reappear if it receives new entries.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public ApiCallable<DeleteLogRequest, Empty> deleteLogCallable() {
+    return ApiUtils.prepareIdempotentCallable(DELETE_LOG, settings).bind(channel);
+  }
+
+  // ----- writeLogEntries -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Writes log entries to Cloud Logging.
+   * All log entries in Cloud Logging are written by this method.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param logName Optional. A default log resource name for those log entries in `entries`
+   * that do not specify their own `logName`.  Example:
+   * `"projects/my-project/logs/syslog"`.  See
+   * [LogEntry][google.logging.v2.LogEntry].
+   * @param resource Optional. A default monitored resource for those log entries in `entries`
+   * that do not specify their own `resource`.
+   * @param labels Optional. User-defined `key:value` items that are added to
+   * the `labels` field of each log entry in `entries`, except when a log
+   * entry specifies its own 'key:value' item with the same key.
+   * Example: `{ "size": "large", "color":"red" }`
+   * @param entries Required. The log entries to write. The log entries must have values for
+   * all required fields.
+   */
+  public WriteLogEntriesResponse writeLogEntries(String logName, MonitoredResource resource, Map<String, String> labels, List<LogEntry> entries) {
+    WriteLogEntriesRequest request =
+        WriteLogEntriesRequest.newBuilder()
+        .setLogName(logName)
+        .setResource(resource)
+        .putAllLabels(labels)
+        .addAllEntries(entries)
+        .build();
+
+    return writeLogEntries(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Writes log entries to Cloud Logging.
+   * All log entries in Cloud Logging are written by this method.
```

Comment:
I think having "All log entries in Cloud Logging are written by this method." is strange when you have 3 of them saying the same.